src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs

```diff
@@ -49,42 +52,54 @@ public IFileProvider FileProvider
         }
 
         /// <summary>
-        /// Gets or sets the configuration name used by <see cref="RazorViewEngine"/> to compile razor views.
+        /// Gets or sets the callback that is used to customize Razor compilation
+        /// to change compilation settings you can update <see cref="RoslynCompilationContext.Compilation"/> property.
+        /// Customizations made here would not reflect in tooling (Intellisense).
         /// </summary>
-        /// <remarks>
-        /// At startup, this is initialized to "Debug" if <see cref="Microsoft.AspNet.Hosting.IHostingEnvironment"/> service is
-        /// registred and environment is development (<see cref="Microsoft.AspNet.Hosting.HostingEnvironmentExtensions.IsDevelopment"/>) else it is set to
-        /// "Release".
-        /// </remarks>
-        public string Configuration
+        public Action<RoslynCompilationContext> CompilationCallback
         {
-            get { return _configuration; }
+            get { return _compilationCallback; }
             set
             {
-                if (string.IsNullOrEmpty(value))
+                if (value == null)
                 {
-                    throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(value));
+                    throw new ArgumentNullException(nameof(value));
                 }
-                _configuration = value;
+                _compilationCallback = value;
             }
         }
 
         /// <summary>
-        /// Gets or sets the callback that is used to customize Razor compilation
-        /// to change compilation settings you can update <see cref="RoslynCompilationContext.Compilation"/> property.
-        /// Customizations made here would not reflect in tooling (Intellisense).
+        /// Gets or sets the <see cref="CSharpParseOptions"/> options used by Razor view compilation.
         /// </summary>
-        public Action<RoslynCompilationContext> CompilationCallback
+        public CSharpParseOptions ParseOptions
```

Comment:
Granted we can make this work, but is there benefit in letting users customize this? Razor parser had to be modified to switch to C# 6, so changing the language version here might not be sufficient.